packages/components/src/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ReactNode, useEffect, useState } from "react";
+import classnames from "classnames";
+import styles from "./Disclosure.css";
+import { Heading } from "../Heading";
+import { Icon } from "../Icon";
+
+export interface DisclosureProps {
+  /**
+   * Child content that is manged by this component.
+   */
+  readonly children: ReactNode | ReactNode[];
+
+  /**
+   * Title for the disclosure pane.
+   */
+  readonly title: string;
+
+  /**
+   * Control if the disclosure component is open or closed.
+   */
+  readonly open?: boolean;
+
+  /**
+   * This sets the default open state of the disclosure.
+   * By default the disclosure is closed.
+   * For use when the component is being used as an Uncontrolled Component.
+   * @default false
+   */
+  readonly defaultOpen?: boolean;
+
+  /**
+   * This function would toggle the open state.
+   * For use when the component is being used as a Controlled Component.
+   */
+  requestOpen?(shouldOpen: boolean): void;
```

Comment:
Sorry, my previous comment was confusing. I was thinking about the argument not the function name. I think keeping the function named `onRequestToggle` probably makes the most sense. :)